modules/tinymce/src/plugins/lists/main/ts/core/Selection.ts

```diff
@@ -48,13 +49,27 @@ const getSelectedListItems = (editor: Editor): Array<HTMLLIElement | HTMLElement
 const getSelectedDlItems = (editor: Editor): HTMLElement[] =>
   Arr.filter(getSelectedListItems(editor), NodeType.isDlItemNode);
 
-const getClosestListRootElm = (editor: Editor, elm: Node): HTMLElement => {
+const getClosestEditingHost = (editor: Editor, elm: Node): HTMLElement => {
   const parentTableCell = editor.dom.getParents<HTMLTableCellElement>(elm, 'TD,TH');
   return parentTableCell.length > 0 ? parentTableCell[0] : editor.getBody();
 };
 
+const isListHost = (schema: Schema, node: Node) => {
+  const nodeName = node.nodeName;
+  const listNames = [ 'UL', 'OL', 'DL' ];
+
+  return !NodeType.isListNode(node) && !NodeType.isListItemNode(node) && Arr.exists(listNames, (listName) => schema.isValidChild(nodeName, listName));
+};
+
+const getClosestListHost = (editor: Editor, elm: Node): HTMLElement => {
+  const parentBlocks = editor.dom.getParents<HTMLElement>(elm, (elm) => editor.dom.isBlock(elm));
+  const parentBlock = Arr.find(parentBlocks, (elm) => isListHost(editor.schema, elm));
+
+  return parentBlock.getOr(editor.getBody());
```

Comment:
Should this be a `getOrThunk` so that it doesn't eagerly evaluate `editor.getBody()`? Granted `editor.getBody()` should be a pretty cheap operation anyways.